File: Leetcode-Medium/167-Two Sum II - Input Array Is Sorted/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l=0,r=numbers.size()-1;
        int n=numbers.size();
        int sum=numbers[l]+numbers[r];
        while(sum!=target)
        {
            if(sum<target)
            {
                
                while(l+1<n and numbers[l+1]==numbers[l])
                {
                    l++;
                }
                l++;
            }
            else
            {
                while(l+1<n and numbers[r-1]==numbers[r])
                {
                    r--;
                }
                r--;
            }
            sum=numbers[l]+numbers[r];
        }
        vector<int> ans;
        ans.push_back(l+1);
        ans.push_back(r+1);
        return ans;
        
    }
};

File: Leetcode-Medium/167-Two Sum II - Input Array Is Sorted/Solution.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i=0
        j=len(numbers)-1
        while i<j:
            s = numbers[i]+numbers[j]
            if s==target: return [i+1,j+1]
            elif s<target: i+=1
            else:j-=1